safe/gis/raster/reclassify.py

```diff
@@ -94,7 +98,10 @@ def reclassify(layer, exposure_key=None, callback=None):
         ranges[hazard_class['value']] = thresholds[hazard_class['key']]
         value_map[hazard_class['key']] = [hazard_class['value']]
 
-    output_raster = unique_filename(suffix='.tiff', dir=temp_dir())
+    if overwrite_input:
+        output_raster = layer.publicSource()
```

Comment:
I think you need to use 'source'.  'publicSource' will have some stars for instance if there is a password.